traveler-swift-core/TravelerKit/Models/CatalogItem.swift

```diff
@@ -17,3 +17,9 @@ public protocol CatalogItem {
     /// URL for a thumbnail
     var imageURL: URL? { get }
 }
+
+/// An item that can be wishlsited
+public protocol WishlistableItem: CatalogItem, Product {
```

Comment:
This should not be a protocol,

Lets talk to API and have them add `isAvailable` to all product JSONs and then just upgrade Product to include `isAvailable`